src/backend/InvenTree/InvenTree/models.py

```diff
@@ -503,200 +502,35 @@ class Meta:
-
-    def __str__(self):
-        """Human name for attachment."""
-        if self.attachment is not None:
-            return os.path.basename(self.attachment.name)
-        return str(self.link)
-
-    attachment = models.FileField(
-        upload_to=rename_attachment,
-        verbose_name=_('Attachment'),
-        help_text=_('Select file to attach'),
-        blank=True,
-        null=True,
-    )
-
-    link = InvenTree.fields.InvenTreeURLField(
-        blank=True,
-        null=True,
-        verbose_name=_('Link'),
-        help_text=_('Link to external URL'),
-    )
-
-    comment = models.CharField(
-        blank=True,
-        max_length=100,
-        verbose_name=_('Comment'),
-        help_text=_('File comment'),
-    )
-
-    user = models.ForeignKey(
-        User,
-        on_delete=models.SET_NULL,
-        blank=True,
-        null=True,
-        verbose_name=_('User'),
-        help_text=_('User'),
-    )
-
-    upload_date = models.DateField(
-        auto_now_add=True, null=True, blank=True, verbose_name=_('upload date')
-    )
+        self.attachments.all().delete()
+        super().delete()
 
     @property
-    def basename(self):
-        """Base name/path for attachment."""
-        if self.attachment:
-            return os.path.basename(self.attachment.name)
-        return None
-
-    @basename.setter
-    def basename(self, fn):
-        """Function to rename the attachment file.
-
-        - Filename cannot be empty
-        - Filename cannot contain illegal characters
-        - Filename must specify an extension
-        - Filename cannot match an existing file
-        """
-        fn = fn.strip()
-
-        if len(fn) == 0:
-            raise ValidationError(_('Filename must not be empty'))
-
-        attachment_dir = settings.MEDIA_ROOT.joinpath(self.getSubdir())
-        old_file = settings.MEDIA_ROOT.joinpath(self.attachment.name)
-        new_file = settings.MEDIA_ROOT.joinpath(self.getSubdir(), fn).resolve()
+    def attachments(self):
+        """Return a queryset containing all attachments for this model."""
+        return self.attachments_for_model().filter(model_id=self.pk)
 
-        # Check that there are no directory tricks going on...
-        if new_file.parent != attachment_dir:
-            logger.error(
-                "Attempted to rename attachment outside valid directory: '%s'", new_file
-            )
-            raise ValidationError(_('Invalid attachment directory'))
-
-        # Ignore further checks if the filename is not actually being renamed
-        if new_file == old_file:
-            return
-
-        forbidden = [
-            "'",
-            '"',
-            '#',
-            '@',
-            '!',
-            '&',
-            '^',
-            '<',
-            '>',
-            ':',
-            ';',
-            '/',
-            '\\',
-            '|',
-            '?',
-            '*',
-            '%',
-            '~',
-            '`',
-        ]
```

Comment:
Q: @SchrodingersGat where was this security feature moved? I am not able to find it in the diff

Reply:
These are largely covered by django's built in file name validators.

Reply:
Are you sure/have you verified this? We had a bunch of vulns in this specific area in the past with length overflows and target directory manipulation and now that huntr is basically dead we wont get a free security audit

Reply:
I will add some more checking in to cover this